gadget/layout.go

```diff
@@ -196,22 +201,25 @@ func layoutVolumeStructures(volume *Volume) (structures []LaidOutStructure, byNa
 		if ps.Name() != "" {
 			byName[ps.Name()] = &ps
 		}
+		if volume.Structure[idx].Offset != nil {
+			offset = *volume.Structure[idx].Offset
+		}
 		// Fill the parts of OnDiskStructure that do not depend on the disk
 		// or on whether we are encrypting or not.
 		// TODO Eventually fill everything here by passing all needed info
 		ps.OnDiskStructure = OnDiskStructure{
 			Name:        ps.VolumeStructure.Name,
 			Type:        ps.VolumeStructure.Type,
-			StartOffset: *volume.Structure[idx].Offset,
+			StartOffset: offset,
```

Comment:
maybe I misremember but we moved some code away from using StartOffset to use Offset but now Offset is not always set, do we have a problem? do we need to recheck all places using Structure.Offset ?